### PR TITLE
Support CLJS on node

### PR DESCRIPTION
**Pull request by DaveWM**
I wanted to be able to use this library in cljs running on node, this PR just makes some small tweaks to the code to allow this. I've also added support for the [doo](https://github.com/bensu/doo) test runner, to test the code works in cljs.

The main changes are in the `reader` files - the clojure code should be unchanged, but they now support cljs as well using different parsing libraries.

### CODE REVIEW

**Comment by immoh** [comment]
Wow, thanks! I will take a look at this later and provide you feedback. Travis build seems to fail now but it looks like it is data generation from spec which has been unstable before.

**Comment by DaveWM** [comment]
No problem :). Think you're right about the travis failure, I did get a couple of failures like that when I was running the tests. I don't think it's related to any of my changes.

File: project.clj [hunk 1]
```diff
@@ -4,7 +4,20 @@
   :license {:name "Eclipse Public License"
             :url "http://www.eclipse.org/legal/epl-v10.html"}
   :dependencies [[org.clojure/clojure "1.9.0-alpha17" :scope "provided"]
+                 [org.clojure/clojurescript "1.9.908"]
```
Comment (immoh) [inline project.clj]: Could add `:scope "provided"` to this? 

(Or alternatively move both clojure & clojurescript dependencies to dev profile.)

File: project.clj [hunk 2]
```diff
@@ -4,7 +4,20 @@
...
                  [org.clojure/spec.alpha "0.1.123"]
                  [circleci/clj-yaml "0.5.5"]
-                 [cheshire "5.7.1"]]
-  :profiles {:dev {:dependencies [[org.clojure/test.check "0.9.0"]]}})
+                 [cheshire "5.7.1"]
+                 [cljs-node-io "0.5.0"]
```
Comment (immoh) [inline project.clj]: I think this is only needed in tests, so could you move it to dev profile dependencies?

**Comment by immoh** [comment]
Looks really good! I will merge this if you fix those two things in project.clj. Tests are also passing locally and only fail on travis so that is ok. I will try to fix that later.

**Comment by immoh** [comment]
It looks like Travis build can be fixed by increasing heap size: https://github.com/immoh/swagger-spec/commit/ef7311180d717a5594a7e0070686f62bb319a888

**Comment by DaveWM** [comment]
Thanks, I've pushed those fixes for project.clj

**Comment by immoh** [comment]
Thanks! I will release new version to Clojars later today.

**Pull request merged**